kroxylicious-kms-provider-hashicorp-vault-test-support/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/AbstractVaultTestKmsFacade.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.util.Set;
+
+import io.kroxylicious.kms.provider.hashicorp.vault.config.Config;
+import io.kroxylicious.kms.service.TestKmsFacade;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public abstract class AbstractVaultTestKmsFacade implements TestKmsFacade<Config, String, VaultEdek> {
+
+    protected static final String VAULT_ROOT_TOKEN = "rootToken";
+    private String kmsVaultToken;
+
+    protected AbstractVaultTestKmsFacade() {
+    }
+
+    protected abstract void startVault();
+
+    protected abstract void stopVault();
+
+    @Override
+    public final void start() {
+
+        startVault();
+
+        // enable transit engine
+        enableTransit();
+
+        // create policy
+        var policyName = "kroxylicious_encryption_filter_policy";
+        try (var policyStream = AbstractVaultTestKmsFacade.class.getResourceAsStream("kroxylicious_encryption_filter_policy.hcl")) {
+            createPolicy(policyName, policyStream);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException("Failed to create policy", e);
+        }
+        // create least-privilege vault token that KMS will use
+        kmsVaultToken = createOrphanToken("kroxylicious_encryption_filter", true, Set.of(policyName));
+    }
+
+    protected abstract void enableTransit();
```

Comment:
It would help readability to have all the abstract methods grouped together I think, maybe underneath `start` since that has most of the logic